app/controllers/companies_controller.rb

```diff
@@ -0,0 +1,62 @@
+class CompaniesController < ApplicationController
+  def index
+    @companies = Company.all
+    if params[:sort] == "location"
+      @companies = @companies.sort_by do |company|
```

Comment:
this is also a lot of logic, sorting a set of `Company` objects could be a method on your model 👍